app/charts/map/map.tsx

```diff
@@ -133,31 +146,30 @@ export const MapComponent = () => {
     bounds,
     bbox,
   } = useChartState() as MapState;
-  const initialViewStateRef = useRef<MinMaxZoomViewState | null>(null);
+
   const isViewStateLocked = controlsType === "locked";
+  // Needed to be able to "refresh" the map to its initial position.
+  const firstViewStateRef = useRef<MinMaxZoomViewState | null>(null);
```

Comment:
Another ref, I really wonder if the logic behind the view states should be extracted in its own hook.